In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [12]:
%matplotlib inline

In [13]:
import warnings
warnings.filterwarnings("ignore")

## **Download dataset**

In [14]:
 !pip install convokit

     |████████████████████████████████| 167 kB 13.1 MB/s 
     |████████████████████████████████| 6.0 MB 50.2 MB/s 
     |████████████████████████████████| 1.5 MB 66.0 MB/s 
     |████████████████████████████████| 235 kB 56.7 MB/s 
     |████████████████████████████████| 175 kB 79.6 MB/s 
     |████████████████████████████████| 53 kB 1.4 MB/s 
     |████████████████████████████████| 749 kB 70.3 MB/s 
     |████████████████████████████████| 451 kB 65.0 MB/s 
     |████████████████████████████████| 10.1 MB 54.1 MB/s 
     |████████████████████████████████| 181 kB 66.9 MB/s 
     |████████████████████████████████| 42 kB 1.0 MB/s 
     |████████████████████████████████| 653 kB 69.3 MB/s 
  Created wheel for convokit: filename=convokit-2.5.3-py3-none-any.whl size=204129 sha256=76e0a6eedb2531d26830272a39365b5232ebd27fe7941030f82dd2f13d2206f2
  Stored in directory: /root/.cache/pip/wheels/b0/e8/2d/81c4477fe586fe4dad2de2886b990e90e839ffccd5158ed0f3
  Created wheel for emoji: filename=emoji-1.7

In [15]:
from convokit import Corpus, download
corpus = Corpus(filename=download("wiki-corpus"))

In [16]:
corpus.print_summary_stats()

Number of Speakers: 38462
Number of Utterances: 391294
Number of Conversations: 125292


## **Use the textual data of the Utterance-level information**

In [17]:
import pandas as pd
import gensim
utt = corpus.get_utterances_dataframe()
display(utt)


timestamp                                               text  \
id                                                                          
524288  1.189190940E09  You should look at all of the point on the tem...   
524289  1.189204860E09  Yes I agree. The law permits usage of document...   
1       1.310744280E09  Yes, that's good. Revathy's page looked very r...   
2       1.310746500E09  Nagma'a site, at least that filmography page, ...   
3       1.310751180E09  I don't think there are many such official rel...   
...                ...                                                ...   
262142  1.285893360E09  Definitely not a waste of time (or at least no...   
393215  1.160237100E09  Hi. [[Empire of Atlantium]] has been [http:\/\...   
393216  1.161765780E09  Hi Debi. Unfortunately I am not in a position ...   
262143  1.285841460E09  Hi Dana. It seems to me that [[User:Yogesh Kha...   
262144  1.285873260E09  At this point, the best way the FAR can move f...   

                           speaker reply_to conversation_id meta.is-admin  \
id                                                                          
524288                   Frightner     None          524288         False   
524289                 Revizionist     None          524288         False   
1                      Johannes003     None               1         False   
2                      Johannes003        1               1         False   
3                      Johannes003        2               1         False   
...                            ...      ...             ...           ...   
262142                 Dana boomer   262141          262137          True   
393215                  Gene Poole     None          393215         False   
393216                       Kappa   393215          393215         False   
262143  The Red Hat of Pat Ferrick     None          262143         False   
262144                 Dana boomer   262143          262143          True   

       vectors  
id              
524288      []  
524289      []  
1           []  
2           []  
3           []  
...        ...  
262142      []  
393215      []  
393216      []  
262143      []  
262144      []  

[391294 rows x 7 columns]

In [ ]:
utt.shape

(391294, 7)

In [ ]:
utt.text

id
524288    You should look at all of the point on the tem...
524289    Yes I agree. The law permits usage of document...
1         Yes, that's good. Revathy's page looked very r...
2         Nagma'a site, at least that filmography page, ...
3         I don't think there are many such official rel...
                                ...                        
262142    Definitely not a waste of time (or at least no...
393215    Hi. [[Empire of Atlantium]] has been [http:\/\...
393216    Hi Debi. Unfortunately I am not in a position ...
262143    Hi Dana. It seems to me that [[User:Yogesh Kha...
262144    At this point, the best way the FAR can move f...
Name: text, Length: 391294, dtype: object

In [ ]:
utt.text[0]

'You should look at all of the point on the template not just the last one, the template also says the image belonging to the republic of macedonia is in the public domain if it being used for \\"information purposes\\". '

## **Simple Data preprocessing**




In [ ]:
#chekcing for one text
gensim.utils.simple_preprocess("You should look at all of the point on the template not just the last one, the template also says the image belonging to the republic of macedonia is in the public domain if it being used for \"information purposes\".")

['you',
 'should',
 'look',
 'at',
 'all',
 'of',
 'the',
 'point',
 'on',
 'the',
 'template',
 'not',
 'just',
 'the',
 'last',
 'one',
 'the',
 'template',
 'also',
 'says',
 'the',
 'image',
 'belonging',
 'to',
 'the',
 'republic',
 'of',
 'macedonia',
 'is',
 'in',
 'the',
 'public',
 'domain',
 'if',
 'it',
 'being',
 'used',
 'for',
 'information',
 'purposes']

In [18]:
#applying it on all the text content
Text1= utt.text.apply(gensim.utils.simple_preprocess)
Text1

id
524288    [you, should, look, at, all, of, the, point, o...
524289    [yes, agree, the, law, permits, usage, of, doc...
1         [yes, that, good, revathy, page, looked, very,...
2         [nagma, site, at, least, that, filmography, pa...
3         [don, think, there, are, many, such, official,...
                                ...                        
262142    [definitely, not, waste, of, time, or, at, lea...
393215    [hi, empire, of, atlantium, has, been, http, e...
393216    [hi, debi, unfortunately, am, not, in, positio...
262143    [hi, dana, it, seems, to, me, that, user, yoge...
262144    [at, this, point, the, best, way, the, far, ca...
Name: text, Length: 391294, dtype: object

## **Using the Gensim word2Vec model to vectorize the text**

In [19]:
model1 = gensim.models.Word2Vec(window= 10, min_count=2, workers=4)

In [20]:
model1.build_vocab(Text1, progress_per=1000)

In [ ]:
model1.epochs

5

In [ ]:
model1.corpus_count

391294

In [21]:
model1.train( Text1, total_examples = model1.corpus_count, epochs=model1.epochs ) 

(66639869, 86231260)

## **Saving the model**

In [22]:
model1.save('my_model1.h5') 

## **Finding the similarities**

In [23]:
model1.wv.most_similar('dana')

[('carlos', 0.7710855603218079),
 ('lordameth', 0.7634602785110474),
 ('vsem', 0.7611789703369141),
 ('cactuswriter', 0.7520367503166199),
 ('colin', 0.7380850315093994),
 ('merbabu', 0.7328872084617615),
 ('matty', 0.731547474861145),
 ('dan', 0.7288472056388855),
 ('falcadore', 0.7242558002471924),
 ('jojan', 0.7241559028625488)]

In [ ]:
model1.wv.most_similar('best')

[('lavigne', 0.5921468734741211),
 ('easiest', 0.5526793003082275),
 ('better', 0.5365970730781555),
 ('good', 0.530225396156311),
 ('avril', 0.5293759107589722),
 ('warmest', 0.5276244282722473),
 ('joyful', 0.5010387897491455),
 ('wise', 0.49304279685020447),
 ('simplest', 0.46616020798683167),
 ('ideal', 0.4379298985004425)]

In [ ]:
model1.wv.most_similar('waste')

[('wasting', 0.8423311114311218),
 ('wasted', 0.7474625110626221),
 ('spend', 0.7320449352264404),
 ('spending', 0.730403482913971),
 ('wastes', 0.6857813596725464),
 ('consuming', 0.6764301657676697),
 ('devote', 0.6599931716918945),
 ('millionth', 0.6163657307624817),
 ('dedicate', 0.6161530017852783),
 ('saver', 0.6039191484451294)]

In [24]:
model1.wv.similarity(w1 = "best" , w2="waste") 

0.08351645

In [25]:
print (list(model1.wv.vocab)) #-----Output is very big, executed this line in ipynb file.

['you', 'should', 'look', 'at', 'all', 'of', 'the', 'point', 'on', 'template', 'not', 'just', 'last', 'one', 'also', 'says', 'image', 'belonging', 'to', 'republic', 'macedonia', 'is', 'in', 'public', 'domain', 'if', 'it', 'being', 'used', 'for', 'information', 'purposes', 'yes', 'agree', 'law', 'permits', 'usage', 'documents', 'photographs', 'and', 'other', 'materials', 'educational', 'informational', 'there', 'was', 'normative', 'act', 'issued', 'by', 'government', 'that', 'even', 'allowed', 'citizens', 'make', 'photocopies', 'or', 'photograph', 'rare', 'archive', 'good', 'page', 'looked', 'very', 'reliable', 'why', 'we', 'as', 'source', 'site', 'least', 'filmography', 'looks', 'like', 'taken', 'from', 'wikipedia', 'itself', 'didn', 'remove', 'time', 'but', 'don', 'think', 'true', 'are', 'many', 'such', 'official', 'sites', 'left', 'yet', 'problem', 'most', 'really', 'often', 'based', 'know', 'what', 'believe', 'madras', 'talkies', 'which', 'dead', 'now', 'http', 'kavithalayaa', 'com'

In [ ]:
#Using nltk k-means cluster-------sample code for my refernce
# from nltk.cluster import KMeansClusterer
#import nltk
#NUM_CLUSTERS=10
#X = model1[model1.wv.vocab]
#kclusterer = KMeansClusterer(NUM_CLUSTERS, distance=nltk.cluster.util.cosine_distance, repeats=25)
#assigned_clusters = kclusterer.cluster(X, assign_clusters=True)
#print (assigned_clusters)

## **Using Sklearn Kmeans cluster function to cluster the created vectors**

In [ ]:
X = model1[model1.wv.vocab]
NUM_CLUSTERS=10

In [ ]:
import numpy as np
from sklearn import cluster
from sklearn import metrics
kmeans = cluster.KMeans(n_clusters=NUM_CLUSTERS)
kmeans.fit(X)
 
labels = kmeans.labels_
centroids = kmeans.cluster_centers_
 
print ("Cluster id labels for inputted data")
print (labels)


Cluster id labels for inputted data
[8 6 9 ... 0 0 0]


## **Grouping the words that are in the same embedding space**

In [ ]:
for i in range(10):
  #model1.wv.most_similar(kmeans.cluster_centers_[i])
  
  print(model1.wv.most_similar([kmeans.cluster_centers_[i]]))

  #model1.most_similar([kmeans.cluster_centers_[i]]) 
  #model1.most_similar([kmeans.cluster_centers_[i]])

[('dbml', 0.9810782670974731), ('vlcsnap', 0.9803299307823181), ('batzuk', 0.9802521467208862), ('imienia', 0.9796416163444519), ('enraptured', 0.9793614745140076), ('zure', 0.97911536693573), ('opor', 0.9788170456886292), ('ovoj', 0.978736162185669), ('aten', 0.9782900214195251), ('spsid', 0.9782158136367798)]
[('za', 0.9398825764656067), ('dat', 0.937229573726654), ('mi', 0.9367580413818359), ('nu', 0.9329489469528198), ('ook', 0.9321944713592529), ('este', 0.9298050403594971), ('gs_sm', 0.9292894005775452), ('outra', 0.9249837398529053), ('treba', 0.9246392846107483), ('bih', 0.9245412945747375)]
[('displaying', 0.721394419670105), ('metadata', 0.7195240259170532), ('taxobox', 0.6854602694511414), ('mediawiki', 0.6824281215667725), ('url', 0.6763370037078857), ('functionality', 0.6756274700164795), ('transcluded', 0.6714636087417603), ('interface', 0.6681926250457764), ('default', 0.6641108989715576), ('vector', 0.6624467372894287)]
[('federation', 0.8631143569946289), ('gardens', 0

In [ ]:
#finding the centroids
print ("Centroids data")
print (centroids)

Centroids data
[[ 1.20239630e-02 -3.93203534e-02  1.88132897e-02  4.64613140e-02
   2.99536269e-02  1.82423200e-02  1.34606548e-02 -6.00385740e-02
  -1.13883987e-04  6.39125556e-02  1.82099156e-02  4.01759446e-02
   4.24219146e-02  1.41447689e-02 -3.90861779e-02 -6.55302405e-02
   4.03646529e-02 -2.23973319e-02  6.64690137e-02  1.83448531e-02
   6.78966288e-03 -4.13068524e-03  8.78467411e-03 -4.24482003e-02
   1.03204869e-01 -3.28546353e-02 -1.08040199e-02  2.81080045e-02
   1.98176205e-02 -1.71532929e-02 -4.64429408e-02  3.24017219e-02
  -9.57935583e-03 -1.33578852e-03 -4.70944718e-02  5.46948388e-02
   3.90987135e-02  2.15496253e-02  3.38760242e-02  2.82930620e-02
  -5.28873503e-03 -8.14045779e-03 -9.67974216e-03  1.06630977e-02
  -6.52413815e-02 -1.32385856e-02  2.94288155e-02 -5.39936200e-02
  -6.88135475e-02 -1.20897321e-02 -1.86970197e-02 -2.50290409e-02
  -5.67237847e-04  5.90711087e-03 -5.01255654e-02  4.73694354e-02
   6.15056381e-02 -1.29321897e-02  4.49463576e-02  1.66573450

In [ ]:
print ("Score (Opposite of the value of X on the K-means objective which is Sum of distances of samples to their closest cluster center):")
print (kmeans.score(X))

Score (Opposite of the value of X on the K-means objective which is Sum of distances of samples to their closest cluster center):
-1016180.5


In [ ]:
silhouette_score = metrics.silhouette_score(X, labels, metric='euclidean')
 
print ("Silhouette_score: ")
print (silhouette_score)

Silhouette_score: 
0.41864872


# **State of Art- FastText**



## **Use the textual data of the Utterance-level information**

In [ ]:
corpus.print_summary_stats()

Number of Speakers: 38462
Number of Utterances: 391294
Number of Conversations: 125292


In [ ]:
#converting into a dataframe
import pandas as pd
import gensim
utt = corpus.get_utterances_dataframe()
display(utt)

timestamp                                               text  \
id                                                                          
524288  1.189190940E09  You should look at all of the point on the tem...   
524289  1.189204860E09  Yes I agree. The law permits usage of document...   
1       1.310744280E09  Yes, that's good. Revathy's page looked very r...   
2       1.310746500E09  Nagma'a site, at least that filmography page, ...   
3       1.310751180E09  I don't think there are many such official rel...   
...                ...                                                ...   
262142  1.285893360E09  Definitely not a waste of time (or at least no...   
393215  1.160237100E09  Hi. [[Empire of Atlantium]] has been [http:\/\...   
393216  1.161765780E09  Hi Debi. Unfortunately I am not in a position ...   
262143  1.285841460E09  Hi Dana. It seems to me that [[User:Yogesh Kha...   
262144  1.285873260E09  At this point, the best way the FAR can move f...   

                           speaker reply_to conversation_id meta.is-admin  \
id                                                                          
524288                   Frightner     None          524288         False   
524289                 Revizionist     None          524288         False   
1                      Johannes003     None               1         False   
2                      Johannes003        1               1         False   
3                      Johannes003        2               1         False   
...                            ...      ...             ...           ...   
262142                 Dana boomer   262141          262137          True   
393215                  Gene Poole     None          393215         False   
393216                       Kappa   393215          393215         False   
262143  The Red Hat of Pat Ferrick     None          262143         False   
262144                 Dana boomer   262143          262143          True   

       vectors  
id              
524288      []  
524289      []  
1           []  
2           []  
3           []  
...        ...  
262142      []  
393215      []  
393216      []  
262143      []  
262144      []  

[391294 rows x 7 columns]

In [ ]:
#retrieving the text
utt.text

id
524288    You should look at all of the point on the tem...
524289    Yes I agree. The law permits usage of document...
1         Yes, that's good. Revathy's page looked very r...
2         Nagma'a site, at least that filmography page, ...
3         I don't think there are many such official rel...
                                ...                        
262142    Definitely not a waste of time (or at least no...
393215    Hi. [[Empire of Atlantium]] has been [http:\/\...
393216    Hi Debi. Unfortunately I am not in a position ...
262143    Hi Dana. It seems to me that [[User:Yogesh Kha...
262144    At this point, the best way the FAR can move f...
Name: text, Length: 391294, dtype: object

In [ ]:
utt.text[0]

'You should look at all of the point on the template not just the last one, the template also says the image belonging to the republic of macedonia is in the public domain if it being used for \\"information purposes\\". '

## **Data preprocessing**

In [ ]:
Text2= utt.text.apply(gensim.utils.simple_preprocess)
Text2

id
524288    [you, should, look, at, all, of, the, point, o...
524289    [yes, agree, the, law, permits, usage, of, doc...
1         [yes, that, good, revathy, page, looked, very,...
2         [nagma, site, at, least, that, filmography, pa...
3         [don, think, there, are, many, such, official,...
                                ...                        
262142    [definitely, not, waste, of, time, or, at, lea...
393215    [hi, empire, of, atlantium, has, been, http, e...
393216    [hi, debi, unfortunately, am, not, in, positio...
262143    [hi, dana, it, seems, to, me, that, user, yoge...
262144    [at, this, point, the, best, way, the, far, ca...
Name: text, Length: 391294, dtype: object

## **Using the Gensim FastText model to vectorize the text**

In [ ]:
model2 = gensim.models.FastText(window= 10, min_count=2, workers=4)

In [ ]:
model2.build_vocab(Text2)

In [ ]:
model2.epochs

5

In [ ]:
model2.corpus_count

391294

In [ ]:
model2.corpus_total_words

17246252

In [ ]:
model2.train(Text2, total_examples=len(Text2), epochs=5) #default epochs is 5

## **Saving the model**

In [ ]:
model2.save('my_model2.h5') 

## **Finding the similarities**

In [ ]:
#finding the similarities
model2.most_similar('dana')

[('danh', 0.9085445404052734),
 ('daneyko', 0.9080114364624023),
 ('kana', 0.9056943655014038),
 ('danah', 0.9031047821044922),
 ('xana', 0.902890682220459),
 ('qana', 0.9010542035102844),
 ('danlina', 0.9003554582595825),
 ('danna', 0.8995572328567505),
 ('ljubljana', 0.8982131481170654),
 ('djana', 0.8962810039520264)]

In [ ]:
model2.most_similar('best')

[('cbest', 0.9352155327796936),
 ('bestbuy', 0.893247127532959),
 ('acbest', 0.8804852962493896),
 ('bestest', 0.8667963147163391),
 ('bestla', 0.840599775314331),
 ('bestimmt', 0.8389848470687866),
 ('beste', 0.8367322087287903),
 ('bestow', 0.83228999376297),
 ('dumbest', 0.8230718374252319),
 ('bestfan', 0.810743510723114)]

In [ ]:
# Finding most similar words
model2.most_similar('waste')

[('wasted_time_r', 0.8588379621505737),
 ('waster', 0.8211139440536499),
 ('wastes', 0.7909239530563354),
 ('wast', 0.7527097463607788),
 ('wasteful', 0.7254390716552734),
 ('wastewater', 0.722235918045044),
 ('wasting', 0.7184856534004211),
 ('waste_treatment', 0.6982898116111755),
 ('wasted', 0.695555567741394),
 ('swastika', 0.68741375207901)]

In [ ]:
model2.similarity(w1 = "best" , w2="waste") 

0.13556185

## **Using Sklearn Kmeans cluster function to cluster the created vectors**

In [ ]:
X = model2[model2.wv.vocab]
NUM_CLUSTERS=10

In [ ]:
from sklearn import cluster
from sklearn import metrics
kmeans = cluster.KMeans(n_clusters=NUM_CLUSTERS)
kmeans.fit(X)
model2.wv.most_similar(centroids)
labels = kmeans.labels_
centroids = kmeans.cluster_centers_
 
print ("Cluster id labels for inputted data")
print (labels)

Cluster id labels for inputted data
[1 9 0 ... 7 8 3]


## **Grouping the words that are in the same embedding space**

In [ ]:
for i in range(10):
  #model1.wv.most_similar(kmeans.cluster_centers_[i])
  
  print(model1.wv.most_similar([kmeans.cluster_centers_[i]]))


[('reoccurrence', 0.3451843857765198), ('duduk', 0.3440002202987671), ('mymcheung', 0.3388846516609192), ('notices', 0.3350279629230499), ('pdcook', 0.32281365990638733), ('realplayer', 0.32140693068504333), ('bband', 0.31727135181427), ('indimidating', 0.316922128200531), ('neptunes', 0.31458809971809387), ('warnings', 0.3126092255115509)]
[('nonsense', 0.3857700526714325), ('messages', 0.37844234704971313), ('silliness', 0.3619897961616516), ('handiwork', 0.3591548502445221), ('vandals', 0.35727620124816895), ('bickering', 0.3542628586292267), ('megabasepairs', 0.353550523519516), ('ip', 0.35098791122436523), ('drama', 0.34536781907081604), ('bband', 0.34398752450942993)]
[('handiwork', 0.3530655801296234), ('unburden', 0.33704060316085815), ('smritokotha', 0.3349987864494324), ('messages', 0.3283268213272095), ('medaltop', 0.3180909752845764), ('evanier', 0.31739872694015503), ('outgrown', 0.31637299060821533), ('mahir', 0.31156858801841736), ('dayan', 0.3041790723800659), ('andd', 

In [ ]:
print ("Centroids data")
print (centroids )

Centroids data
[[ 1.03594344e-02 -3.77049930e-02  1.48017481e-02  4.17364202e-02
   2.53349096e-02  1.67586152e-02  1.35143530e-02 -5.10225855e-02
  -1.95429288e-03  5.36445081e-02  1.35421939e-02  3.39622274e-02
   3.80639657e-02  1.51207102e-02 -3.50179039e-02 -5.89188859e-02
   3.36106867e-02 -1.94559097e-02  6.18168116e-02  1.65417995e-02
   8.05132184e-03 -2.90922262e-03  8.22572410e-03 -3.67901251e-02
   8.94191712e-02 -2.74847709e-02 -7.45652243e-03  2.49838792e-02
   1.82687864e-02 -1.44742429e-02 -3.97694446e-02  2.74077021e-02
  -8.26709159e-03 -4.49099904e-03 -4.21246551e-02  4.91925627e-02
   3.08848880e-02  2.07260847e-02  3.10594514e-02  2.30401028e-02
  -6.20983914e-03 -5.75157534e-03 -9.22362693e-03  8.72356631e-03
  -5.58797792e-02 -1.06452564e-02  2.95084827e-02 -4.79539335e-02
  -6.01069219e-02 -1.07741226e-02 -1.69412978e-02 -2.19579302e-02
  -3.53471283e-03  4.15563956e-03 -4.48833443e-02  3.94153595e-02
   5.55326641e-02 -8.85271654e-03  3.98015417e-02  1.51654221

In [ ]:
print ("Score (Opposite of the value of X on the K-means objective which is Sum of distances of samples to their closest cluster center):")
print (kmeans.score(X))

Score (Opposite of the value of X on the K-means objective which is Sum of distances of samples to their closest cluster center):
-1015331.0625


###**Applied the K-means clustering on both word2vec and FastText model. The result is as shown below. Fasttext model retrieve the relevant words compare to word2 vec. In this scenario, Word2vec model is faster than FastText.Changing the word embedding will definetely affect the cluster performance. More relevant words are found in fatstext. In Word2vec, it treats the smallest words to train on it. Whereas FastText will treat each word as composed of character n-grams**


# ** Output of Word2vec upon K-means clustering**


[('dbml', 0.9810782670974731), ('vlcsnap', 0.9803299307823181), ('batzuk', 0.9802521467208862), ('imienia', 0.9796416163444519), ('enraptured', 0.9793614745140076), ('zure', 0.97911536693573), ('opor', 0.9788170456886292), ('ovoj', 0.978736162185669), ('aten', 0.9782900214195251), ('spsid', 0.9782158136367798)]
[('za', 0.9398825764656067), ('dat', 0.937229573726654), ('mi', 0.9367580413818359), ('nu', 0.9329489469528198), ('ook', 0.9321944713592529), ('este', 0.9298050403594971), ('gs_sm', 0.9292894005775452), ('outra', 0.9249837398529053), ('treba', 0.9246392846107483), ('bih', 0.9245412945747375)]
[('displaying', 0.721394419670105), ('metadata', 0.7195240259170532), ('taxobox', 0.6854602694511414), ('mediawiki', 0.6824281215667725), ('url', 0.6763370037078857), ('functionality', 0.6756274700164795), ('transcluded', 0.6714636087417603), ('interface', 0.6681926250457764), ('default', 0.6641108989715576), ('vector', 0.6624467372894287)]
[('federation', 0.8631143569946289), ('gardens', 0.8610069751739502), ('sweden', 0.8589562773704529), ('spain', 0.8555359840393066), ('kentucky', 0.8536316156387329), ('scotland', 0.8530076742172241), ('denmark', 0.8447362780570984), ('austria', 0.8392874598503113), ('philippines', 0.8367154598236084), ('kings', 0.8354343175888062)]
[('elements', 0.7579073905944824), ('characteristics', 0.7464014291763306), ('definitions', 0.7460722327232361), ('concepts', 0.7284207344055176), ('terms', 0.7069368958473206), ('comparisons', 0.6904000043869019), ('meanings', 0.6836440563201904), ('variations', 0.6822459697723389), ('methods', 0.6766904592514038), ('forms', 0.6716595888137817)]
[('odrodzenie', 0.9479486346244812), ('oznacavalo', 0.9445313215255737), ('procesy', 0.9415119886398315), ('polsce', 0.9400675892829895), ('razliku', 0.938963770866394), ('crveno', 0.935950756072998), ('dokumenti', 0.9337365031242371), ('rata', 0.9330689907073975), ('albanci', 0.932388186454773), ('izdati', 0.9323278069496155)]
[('irrelevant', 0.7723680734634399), ('meaningless', 0.7443519830703735), ('subjective', 0.7144993543624878), ('defining', 0.7108522653579712), ('presented', 0.6970717906951904), ('pseudoscience', 0.6888107061386108), ('purely', 0.6885997653007507), ('defined', 0.6841267347335815), ('significant', 0.6825767755508423), ('misleading', 0.6787734031677246)]
[('finishing', 0.7297284007072449), ('stuck', 0.7175493240356445), ('kicked', 0.7013392448425293), ('caught', 0.6998166441917419), ('pissed', 0.6883466243743896), ('kick', 0.6751716136932373), ('sitting', 0.6720247268676758), ('asleep', 0.670079231262207), ('walked', 0.6637715101242065), ('blew', 0.6624693870544434)]
[('actions', 0.8121183514595032), ('behavior', 0.7986999750137329), ('behaviour', 0.7722811698913574), ('conduct', 0.7538747787475586), ('complaint', 0.7487471699714661), ('incivility', 0.7383655309677124), ('harassment', 0.7296298742294312), ('sanction', 0.7231175303459167), ('accusation', 0.7229013442993164), ('disruption', 0.7108350396156311)]
[('defer', 0.7237309217453003), ('demonstrate', 0.7167040109634399), ('argue', 0.6947859525680542), ('convince', 0.6940929889678955), ('accept', 0.6938302516937256), ('occur', 0.6926417350769043), ('alter', 0.6906852722167969), ('involve', 0.6749091148376465), ('add', 0.6725741624832153), ('relate', 0.6688225269317627)]

## **Output of FastText applied K-means clustering**
[('reoccurrence', 0.3451843857765198), ('duduk', 0.3440002202987671), ('mymcheung', 0.3388846516609192), ('notices', 0.3350279629230499), ('pdcook', 0.32281365990638733), ('realplayer', 0.32140693068504333), ('bband', 0.31727135181427), ('indimidating', 0.316922128200531), ('neptunes', 0.31458809971809387), ('warnings', 0.3126092255115509)]
[('nonsense', 0.3857700526714325), ('messages', 0.37844234704971313), ('silliness', 0.3619897961616516), ('handiwork', 0.3591548502445221), ('vandals', 0.35727620124816895), ('bickering', 0.3542628586292267), ('megabasepairs', 0.353550523519516), ('ip', 0.35098791122436523), ('drama', 0.34536781907081604), ('bband', 0.34398752450942993)]
[('handiwork', 0.3530655801296234), ('unburden', 0.33704060316085815), ('smritokotha', 0.3349987864494324), ('messages', 0.3283268213272095), ('medaltop', 0.3180909752845764), ('evanier', 0.31739872694015503), ('outgrown', 0.31637299060821533), ('mahir', 0.31156858801841736), ('dayan', 0.3041790723800659), ('andd', 0.3037582039833069)]
[('unburden', 0.35258960723876953), ('messages', 0.3437122702598572), ('assortment', 0.3372906446456909), ('rubish', 0.31944578886032104), ('outside_view', 0.3168724775314331), ('wikiptroject', 0.3119719326496124), ('quickie', 0.3045141100883484), ('reoccurrence', 0.3042176365852356), ('delinking', 0.30322545766830444), ('onefortyone', 0.30177783966064453)]
[('messages', 0.3500863313674927), ('chemmani', 0.3301903307437897), ('unnessary', 0.32985302805900574), ('ip', 0.32791954278945923), ('fibroceutical', 0.32753604650497437), ('notices', 0.3219844400882721), ('reoccurrence', 0.32027676701545715), ('ascherse', 0.31892597675323486), ('inoperable', 0.3185462951660156), ('megabasepairs', 0.31580406427383423)]
[('daumier', 0.33922073245048523), ('piotrus_', 0.33376339077949524), ('mymcheung', 0.3333449065685272), ('edubuntu', 0.328385591506958), ('messages', 0.32417169213294983), ('herpetology', 0.3163972795009613), ('reoccurrence', 0.3162253201007843), ('abahlali', 0.3156106472015381), ('posting', 0.3126547932624817), ('received', 0.3092849850654602)]
[('uncalled', 0.33684855699539185), ('dyk_for_gray', 0.336269736289978), ('rubish', 0.3332090973854065), ('urbanconnects', 0.33196860551834106), ('mymcheung', 0.32363417744636536), ('puerile', 0.31589746475219727), ('outgrown', 0.3153540790081024), ('wipipedia', 0.3143012225627899), ('messages', 0.3140455484390259), ('outside_view', 0.30863749980926514)]
[('inappropriate', 0.3346952497959137), ('duduk', 0.33053070306777954), ('materialist', 0.3164226710796356), ('umpteenth', 0.31550124287605286), ('purpose', 0.31535646319389343), ('dismaying', 0.30840134620666504), ('messages', 0.3076658248901367), ('delinking', 0.3014424741268158), ('courtesy', 0.30123281478881836), ('inflamatory', 0.29787278175354004)]
[('messages', 0.33221226930618286), ('delinking', 0.329099178314209), ('megabasepairs', 0.3246414363384247), ('memoserv', 0.3237157464027405), ('duduk', 0.3162825107574463), ('mymcheung', 0.3115370571613312), ('courtesy', 0.3058967888355255), ('notices', 0.30388104915618896), ('purpose', 0.3003356456756592), ('umpteenth', 0.2979872226715088)]
[('messages', 0.3448660969734192), ('unburden', 0.34205713868141174), ('assortment', 0.3330410122871399), ('handiwork', 0.33127453923225403), ('applauding', 0.3265150785446167), ('prompting', 0.32573479413986206), ('ascherse', 0.3229474425315857), ('ip', 0.3188170790672302), ('stenton', 0.31491127610206604), ('vitriol', 0.3144073784351349)]

In [ ]:
%%shell
jupyter nbconvert --to html /content/drive/MyDrive/BigData/wordEmbedding.ipynb 

[NbConvertApp] Converting notebook /content/drive/MyDrive/BigData/wordEmbedding.ipynb to html
[NbConvertApp] Writing 390826 bytes to /content/drive/MyDrive/BigData/wordEmbedding.html
